_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp

```diff
@@ -515,12 +515,12 @@ mfxU16 MakeSlices(MfxVideoParam& par, mfxU32 SliceStructure)
     if (par.m_ext.CO2.NumMbPerSlice != 0)
         nSlice = CeilDiv(nLCU / nTile, par.m_ext.CO2.NumMbPerSlice) * nTile;
 
+    if (nSlice >= 1)
```

Comment:
@fulinjie , sorry, I didn't get: why line #505 doesn't set correct num of slices?

Reply:
That's because par.m_platform is MFX_HW_ICL_LP. (MFX_HW_ICL_LP = MFX_HW_ICL + 1)

Actually, I can see all the ICL devices listed in listLegalDevIDs would be registered as MFX_HW_ICL_LP.

As a consequence, is it more reasonable to change into:
if (par.m_platform <= MFX_HW_ICL_LP)

Reply:
OMG

Reply:
Please re-work the PR as the attached one 
[0001-Slice-num-fix.txt](https://github.com/Intel-Media-SDK/MediaSDK/files/3895888/0001-Slice-num-fix.txt)